### PR TITLE
Update Prettier Solidity 

### PR DESCRIPTION
**Pull request by frangio**
The new version of Prettier Solidity changes a few things about how it formats the code, so this PR includes many small changes to formatting.

I wrote a little program to check that files in this PR didn't change in their actual Solidity content, other than comments and whitespace: [solidity-fingerprint](https://github.com/frangio/solidity-fingerprint).

### CODE REVIEW

**Review by ernestognw — APPROVED** [review]
Ran tests for both master and this branch.
Everything seems fine 👍 

[GIst](https://gist.github.com/ernestognw/f27c226c46928ab88a0b36a47812a6f0.js)

File: CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,7 @@
 
 ## Unreleased
 
+ * Reformatted codebase with latest version of Prettier Solidity. ([#3898](https://github.com/OpenZeppelin/openzeppelin-contracts/pull/3898))
```
[the remark this answers is not in the document]
Comment (frangio) [reply CHANGELOG.md:5]: Normally formatting changes are not included in the changelog, but these are a lot of formatting changes so I thought it should be mentioned given that the versioned headers will get updated.

**Comment by Janther** [comment]
there are a few scenarios where `prettier-plugin-solidity` changes the code to improve readability. However we make sure these changes don't affect the compiled code.
For example `a * b / c` will be formatted as `(a * b) / c` or `a ** b ** c` will result in `a ** (b ** c)`.
these changes will generate a different fingerprint in your program since we added `()` but the compiled code will be the same.
So in the future, when we release a new version, you might want to consider comparing compiled codes instead.

**Comment by frangio** [comment]
@Janther The issue is that things like abstract contracts don't produce bytecode. But you're right that the fingerprint as I defined it here is not the right approach. Instead of computing a fingerprint of the token stream, it should probably be of the AST.

**Pull request merged**